server/src/test/java/org/elasticsearch/cluster/action/shard/ShardStartedClusterStateTaskExecutorTests.java

```diff
@@ -188,18 +170,18 @@ public void testStartedShards() throws Exception {
                 )
             );
         }
-        final ClusterStateTaskExecutor.ClusterTasksResult<?> result = executeTasks(clusterState, tasks);
-        assertNotSame(clusterState, result.resultingState());
-        assertThat(result.executionResults().size(), equalTo(tasks.size()));
```

Comment:
I think size assertion is missing in the updated code. Was it intentional?

Reply:
Aha, it is no longer available in result. Disregard above comment

Reply:
Yeah there's no way to break this any more: it would require changing the list of task contexts, but this list comes from `Stream#toList()` so it's immutable. Really here the original code was doing this to verify that the set of completed tasks was exactly what we expected, but we check that directly now.